src/include/duckdb/planner/operator/logical_limit.hpp

```diff
@@ -25,6 +25,8 @@ class LogicalLimit : public LogicalOperator {
 	//! The offset from the start to begin emitting elements
 	unique_ptr<Expression> offset;
 
+	bool force_stream_and_parallel = false;
```

Comment:
While I understand the intention behind the flag - I feel it is a bit of a premature optimization that is out of scope for this PR. Could we remove this flag and leave only the optimizer fix?